File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float owed;
    // Obtain change amount (non-negative)
    do
    {
        owed = get_float("Change owed: ");
    }
    while (owed < 0);

    // Convert dollars to cents to avoid floating-point imprecision errors
    int cents = round(owed * 100);

    // Set up counter to deterine how many coins given
    int counter = 0;

    // While loops to subtract largest coints first based on amount of change left
    while (cents >= 25)
    {
        cents -= 25;
        counter += 1;
    }

    while (cents >= 10 && cents < 25)
    {
        cents -= 10;
        counter += 1;
    }

    while (cents >= 5 && cents < 10)
    {
        cents -= 5;
        counter += 1;
    }

    while (cents > 0 && cents < 5)
    {
        cents -= 1;
        counter += 1;
    }

    printf("%i\n", counter);
}

File: pset6/sentimental-cash/cash.py
from cs50 import get_float

# Obtain change amount (non-negative number)
change_amount = get_float("Change owed: ")
while change_amount < 0:
    change_amount = get_float("Change owed: ")

# Counter for number of coins given
coins = 0

# Iterate through change amount and decrease starting with quarters, then dimes, etc.
while change_amount >= .25:
    change_amount -= .25
    coins += 1
while change_amount >= .10 and change_amount < .25:
    change_amount -= .10
    coins += 1
while change_amount >= .05 and change_amount < .10:
    change_amount -= .05
    coins += 1
while change_amount < .05 and change_amount > 0:
    change_amount -= .01
    coins += 1

# Print number of coins
print(coins)